### PR TITLE
Add navigability column to hotspots list

### LINKED ISSUE #1804: Display navigability status in the hotspots list
e.g. can't find the file
can't find the issue in the file

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Fixes #1804 

image

### CODE REVIEW

File: src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/NavigabilityTableColumnDefinition.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        public override string DisplayName { get; } = "";
+
+        public override double DefaultWidth { get; } = Width;
+
+        public override bool IsFilterable { get; } = false;
+
+        public override bool IsSortable { get; } = false;
+
+        public override bool TryCreateToolTip(ITableEntryHandle entry, out object toolTip)
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/NavigabilityTableColumnDefinition.cs:47]: I preferred to place the entire navigability logic here, instead of in the general HotspotTableEntry. This way the logic is isolated and HotspotTableEntry doesn't repeatedly call these methods for the wrong columns.

File: src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs [hunk 2]
```diff
@@ -29,13 +29,12 @@ namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableD
     internal class PriorityTableColumnDefinition : TableColumnDefinitionBase
     {
         public const string ColumnName = "ReviewPriority";
+        public const int Width = 120;
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs:32]: Minor change: the column was a bit too small for the current values

File: src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs [hunk 3]
```diff
@@ -29,13 +29,12 @@ namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableD
...
 
         public override string Name { get; } = ColumnName;
 
-        public override double DefaultWidth => 50;
+        public override string DisplayName { get; } = "Priority";
 
-        public override string DisplayName => "Priority";
-
-        public override bool IsFilterable => true;
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs:39]: IsFilterable=true is the default value, so no need to specify it.

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableColumns.cs [hunk 4]
```diff
@@ -29,8 +29,9 @@ internal static class HotspotsTableColumns
     {
         public static IReadOnlyList<ColumnState> InitialStates { get; } = new[]
         {
-            new ColumnState(StandardTableColumnDefinitions.ErrorCode, true, 50),
-            new ColumnState(PriorityTableColumnDefinition.ColumnName, true, 100),
+            new ColumnState(NavigabilityTableColumnDefinition.ColumnName, true, NavigabilityTableColumnDefinition.Width),
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableColumns.cs:32]: No copy-paste of the width

File: src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs [hunk 5]
```diff
@@ -29,13 +29,12 @@ namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableD
     internal class PriorityTableColumnDefinition : TableColumnDefinitionBase
     {
         public const string ColumnName = "ReviewPriority";
+        public const int Width = 120;
 
         public override string Name { get; } = ColumnName;
 
-        public override double DefaultWidth => 50;
+        public override string DisplayName { get; } = "Priority";
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs:36]: Minor unrelated change: changed to single initialization as the value will never change.

File: src/IssueViz.Security/Resources.resx [hunk 6]
```diff
@@ -117,6 +117,9 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="ERR_CannotNavigate" xml:space="preserve">
+    <value>Cannot navigate to step. The source code is different from the analyzed version</value>
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/Resources.resx:121]: I think `Location` would be better than `step`, and since there is more than one sentence the second sentence should have a full stop tool.
Also, it helps to prefix the resource name with the component that uses it, particularly since this is in the top-level resource file.

File: src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs (hunk 3 above)
Comment (duncanp-sonar) [reply src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs:39]: Perhaps not from a functionality point of view, but it doesn't help the reader unless they happen to know it is the default...

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableColumns.cs (hunk 4 above)
Comment (duncanp-sonar) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableColumns.cs:32]: I don't understand your comment.

Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableColumns.cs:32]: The width value was copy-pasted: here and in the column definition. So now I simply reference the value in the column definition.

File: src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs (hunk 3 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/PriorityTableColumnDefinition.cs:39]: There are other properties, such as IsSortable, IsResizable, IsCopyable - it's noisy to repeat them and just set to default values.

File: src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/NavigabilityTableColumnDefinition.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/NavigabilityTableColumnDefinition.cs:47]: Localising the code makes sense, but why would the HotspotTableEntry "repeatedly call these methods for the wrong columns"?

Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/CustomColumns/NavigabilityTableColumnDefinition.cs:47]: It checks for each column if it has a tooltip/image. Currently the base class has an empty implementation for them.
If I were to put this code in the hotspots table, then it'd have to have a break `if (columnName != "Navigability column name") return;`. Just noisy. This is cleaner.

**Review by duncanp-sonar — APPROVED** [review]
